### ISSUE TITLE
Keep the field Username visible when selecting External Tool

### ISSUE THREAD

**Issue by Alessio71** [mRemoteNG/mRemoteNG#2196]
## Expected Behavior

In previous versions the field Username was always visible, even when changing Protocol to External Tool

## Current Behavior

With latest nightly the field get removed when changing the Protocol, for example, from RDP to External Tool

## Possible Solution

Keep it visible

## Steps to Reproduce (for bugs)

1. Create a new connection
2. Choose External Tool as Protocol
3. The Username field disappears 

## Your Environment

* Version used:
* Operating System and version (e.g. Windows 10 1709 x64):
Nightly build  2022.01.07-1.77.2-nb
on
Windows 11 x64

**Comment by Alessio71** [comment]
Username appeared even when External tool is selected, issue fixed. thank you.
Will the password without focus by default be address too?
Thank you very much for your help.
Alex

**Comment by Kvarkas** [comment]
I think We fix autofocus, could you check last NB please?

**Comment by Alessio71** [comment]
On build 3208 is not working
Thanks,
*Alessio*

**Comment by Alessio71** [comment]
> I think We fix autofocus, could you check last NB please?
@Kvarkas,
autofocus Is still not working on opening on 1.78.2.3228, after the splash screen disappears, the unlock password request has no focus.
Wasn't working on Windows 11 x64 24H2, still not working now on 25H2.
Is not a big issue anyway, all the rest is working just fine
Thank you for your work,
Alessio

**Comment by Kvarkas** [comment]
hello, could you mention please a steps how to reproduce as maybe i am looking to another place? few printscreens also welcome

**Comment by Alessio71** [comment]
> hello, could you mention please a steps how to reproduce as maybe i am looking to another place? few printscreens also welcome

I'll try.
notepad opened in background, double click on mRemote icon

<img width="627" height="432" alt="Image" src="https://github.com/user-attachments/assets/aff85924-ad8f-4121-b568-043719e2a63e" />

after the logo disappears the password request appears but the focus is back on notepad
 
<img width="761" height="523" alt="Image" src="https://github.com/user-attachments/assets/efda82b2-403b-4255-80b6-a5313f20faca" />

Can you reproduce the issue following these steps?
thank you,
Alessio

**Comment by robertpopa22** [comment]
**Fix available for testing**

A fix for this issue has been implemented in our community fork (`robertpopa22/mRemoteNG`, branch `main`). A beta build including this fix is available from the fork's Releases page.

Please test and report if this resolves your issue.

---

> *We sincerely appreciate the mRemoteNG maintainers and the wonderful community that has kept this project alive and thriving for over a decade. Your work powers thousands of IT professionals worldwide, including us. Our fork builds entirely on your foundation — we're preparing a consolidated Pull Request to contribute these improvements back to the official project. Thank you for everything you've built.*
>
> *— robertpopa22, community contributor*

**Comment by Alessio71** [comment]
Hi @robertpopa22, I don't understand which version you're referring to but latest 20260225 (v1.81.0-beta.4) (that appears as 1.80.2 3410 on About) the issue is not resolved, after the logo disappares the password request doesn't have focus.

Thank you

**Comment by Alessio71** [comment]
This new version brings new problems too, when I launch a connection that uses an external tool, this doesn't start but appears a new tab like this

<img width="845" height="482" alt="Image" src="https://github.com/user-attachments/assets/b6bc28c2-2811-444d-ad4d-43589c8cd9ea" />

and when you try and close it, this:

<img width="910" height="703" alt="Image" src="https://github.com/user-attachments/assets/d33c28f3-e796-473c-acc9-e4a9c209ae33" />

**Comment by robertpopa22** [comment]
Still some 200+ issues to solve from backlog, and then we close this also! Just a couple of days, pls.

---

> *We sincerely appreciate the mRemoteNG maintainers and the wonderful community that has kept this project alive and thriving for over a decade. Your work powers thousands of IT professionals worldwide, including us. Our fork builds entirely on your foundation — we're preparing a consolidated Pull Request to contribute these improvements back to the official project. Thank you for everything you've built.*
>
> *— robertpopa22, community contributor*

**Comment by robertpopa22** [comment]
THank you for helping with tests!

● IIS Orchestrator Status: RUNNING                          Report: 2026-02-25 16:40
  Phase: test_hygiene (pre-flight baseline) — running parallel tests

  --- Processes --------------------------------------------------
    PID   Start     CPU  Mem   Role
    49500 16:37:18  0.8  27M   Supervisor (orchestrator_supervisor.py)
    50496 16:37:59  0.4  34M   Orchestrator (iis_orchestrator.py --force)
    Health: OK

  Session (started 16:37):
    Build: OK (16:38:46)
    Tests: running 9 groups (Phase 1, max 2 concurrent)
    Issues: not started yet (test hygiene first)
    Commits: 0 | Cost: $0.00 | Duration: 00:03

  --- Overall Progress (all sessions) ----------------------------
  Issues DB: 839 total
    DB Status:  Released: 502 | Testing: 5 | Triaged: 99 | New: 149
                Wontfix: 56 | Duplicate: 24 | Needs Human: 4

    Verified (cross-referenced with git):
      Issues with fix commits: 475 unique
      Closed without code:      80 (wontfix + duplicate)
      Truly resolved:          548/839
      [███████████████████░░░░░░░░░░░] 65%

    Remaining: 291
      Testing (post-beta.4):   5
      Triaged (no commit):    99
      New (not started):     149
      Needs human:             4

    Calculation: 475 + 80 - 7 (double-counted) = 548 resolved
                 839 - 548 = 291 remaining

    Status tracking: OK (0 gap)

  Tests: 2927 total

  --- New Issues (last 7 days) -----------------------------------
  Upstream (mRemoteNG/mRemoteNG):
    #3173 Possible command injection via Process.Start [AI, critical]
    #3170 Opening Command sent too soon during SSH login [SSH, Improvement]
    #3167 mRemoteNG 1.78.2 NB 3405 (SC) requires .NET 9.0
    #3166 Jump-host tabs wrong resolution after RDP reconnect [RDP, UI/UX]
    #3163 белый экран [Bug]

  Fork (robertpopa22/mRemoteNG):
    #20 [Crash] System.InvalidCastException: ConnectionInfo cast

  Orchestratorul rulează — tocmai a trecut build-ul și acum face test hygiene pre-flight (baseline
  tests). După asta va trece la issues.

---

> *We sincerely appreciate the mRemoteNG maintainers and the wonderful community that has kept this project alive and thriving for over a decade. Your work powers thousands of IT professionals worldwide, including us. Our fork builds entirely on your foundation — we're preparing a consolidated Pull Request to contribute these improvements back to the official project. Thank you for everything you've built.*
>
> *— robertpopa22, community contributor*

**Comment by robertpopa22** [comment]
**Fix available for testing**

A fix for this issue has been implemented in our community fork (`robertpopa22/mRemoteNG`, branch `main`). A beta build including this fix is available from the fork's Releases page.

Please test and report if this resolves your issue.

---

> *We sincerely appreciate the mRemoteNG maintainers and the wonderful community that has kept this project alive and thriving for over a decade. Your work powers thousands of IT professionals worldwide, including us. Our fork builds entirely on your foundation — we're preparing a consolidated Pull Request to contribute these improvements back to the official project. Thank you for everything you've built.*
>
> *— robertpopa22, community contributor*

**Comment by Alessio71** [comment]
Hi @robertpopa22, no problem.
BTW, this is what I see from your messages when they points to a repository

<img width="922" height="170" alt="Image" src="https://github.com/user-attachments/assets/bff88548-c10f-4847-a7a0-555dbc3ea9ef" />

**Comment by Alessio71** [comment]
I've tried Nightly Build — 20260225 (v1.81.0-beta.4 a bit better, I was launching my external tools with .\ prefix to the command; removing it now they launch correctly, only the weird tab remains but when you close it now disappears without errors (maybe because it managed to launch the external tool correctly).
The focus on password request after logo works better too, doesn't get the focus only when launched from the taskbar pinned app.
Thank you!

**Comment by Alessio71** [comment]
I have to correct what I wrote, the launching of external tools works only once, every time I have to close the weird tab to make it work again (it's  a bit frustrating)

**Comment by Alessio71** [comment]
BTW, this tab

<img width="609" height="341" alt="Image" src="https://github.com/user-attachments/assets/092aa4c2-7071-4b88-886f-b41aa68bc1f6" />

and then this

<img width="608" height="474" alt="Image" src="https://github.com/user-attachments/assets/a7c05c73-2d4b-43b4-aae4-a93ed4eaf526" />

when you close it, appears even clicking once on the Connection entry

It appears even with connection that are using native RDP connection (not only the one with External tool)

<img width="598" height="336" alt="Image" src="https://github.com/user-attachments/assets/29c88c45-ebf1-4240-9ae7-1281b5d353b6" />

**Comment by robertpopa22** [comment]
**Fix available for testing**

**Commit:** `dae524af` (fork `robertpopa22/mRemoteNG`, branch `main`)
**What changed:** Keep the field Username visible when selecting External Tool

A beta build (v1.81.0-beta.4) including this fix is available from the fork's Releases page.

---

> *We sincerely appreciate the mRemoteNG maintainers and the wonderful community that has kept this project alive and thriving for over a decade. Your work powers thousands of IT professionals worldwide, including us. Our fork builds entirely on your foundation — we're preparing a consolidated Pull Request to contribute these improvements back to the official project. Thank you for everything you've built.*
>
> *— robertpopa22, community contributor*

**Comment by Alessio71** [comment]
I'm using v1.81.0-beta.4, password is now visible when external tool is selected, I confirm that that issue is fixed.
Remains the (minor) focus on password issue when mRemoteNG is launched as a pined app from the taskbar (the focus is working when launched from a link from the desktop, for example) and the new issue about the annoying tab that opens when clicking on every connection and that prevents the connection if you don't close it and re double click the (same) connection once more.
Thank you.
